cardano-node/src/Cardano/CLI/Ops.hs

```diff
@@ -221,3 +272,151 @@ withRealPBFT gHash genFile nMagic sigThresh delCertFp sKeyFp update ptcl action
+                        Left err -> do putTextLn . toS $ show err
+                                       exitFailure
+
+  createNodeConnection (Proxy) p sockPath
+
+
+createNodeConnection
+  :: forall blk . (Condense (HeaderHash blk), RunNode blk)
+  => Proxy blk
+  -> Consensus.Protocol blk
+  -> SocketPath
+  -> IO ()
+createNodeConnection proxy ptcl socketPath = do
+    addr <- localSocketAddrInfo socketPath
+    let ProtocolInfo{pInfoConfig} = protocolInfo ptcl
+    connectTo
+      (NetworkConnectTracers nullTracer nullTracer)
+      (localInitiatorNetworkApplication proxy pInfoConfig)
+      Nothing
+      addr
+    `catch` handleMuxError
+
+handleMuxError :: MuxError -> IO ()
+handleMuxError err = print err
+
+localInitiatorNetworkApplication
+  :: forall blk m peer.
+     ( RunNode blk
+     , Condense (HeaderHash blk)
+     , MonadIO m
+     , MonadST m
+     , MonadThrow m
+     , MonadTimer m
+     )
+  => Proxy blk
+  -> Consensus.NodeConfig (BlockProtocol blk)
+  -> Versions NodeToClientVersion DictVersion
+              (OuroborosApplication 'InitiatorApp peer NodeToClientProtocols
+                                    m LB.ByteString () Void)
+localInitiatorNetworkApplication proxy pInfoConfig =
+    simpleSingletonVersions
+      NodeToClientV_1
+      (NodeToClientVersionData { networkMagic = nodeNetworkMagic proxy pInfoConfig })
+      (DictVersion nodeToClientCodecCBORTerm)
+
+  $ OuroborosInitiatorApplication $ \_peer ptcl -> case ptcl of
+      LocalTxSubmissionPtcl -> \channel ->
+        runPeer
+          nullTracer
+          (localTxSubmissionCodec proxy)
+          channel
+          (localTxSubmissionClientPeer localTxSubmissionClientNull)
+
+      ChainSyncWithBlocksPtcl -> \channel ->
+        runPeer
+          nullTracer
+          (localChainSyncCodec @blk pInfoConfig)
+          channel
+          (chainSyncClientPeer chainSyncClient)
+
+localChainSyncCodec
+  :: forall blk m . (RunNode blk, MonadST m)
+  => Consensus.NodeConfig (BlockProtocol blk)
+  -> Codec (ChainSync blk (Tip blk))
+           DeserialiseFailure m LB.ByteString
+localChainSyncCodec pInfoConfig =
+  codecChainSync
+    (wrapCBORinCBOR (nodeEncodeBlock pInfoConfig))
+    (unwrapCBORinCBOR (nodeDecodeBlock pInfoConfig))
+    (encodePoint (nodeEncodeHeaderHash (Proxy @blk)))
+    (decodePoint (nodeDecodeHeaderHash (Proxy @blk)))
+    (encodeTip (nodeEncodeHeaderHash (Proxy @blk)))
+    (decodeTip (nodeDecodeHeaderHash (Proxy @blk)))
+
+localTxSubmissionCodec
+  :: forall m blk . (RunNode blk, MonadST m)
+  => Proxy blk
+  -> Codec (LocalTxSubmission (GenTx blk) (ApplyTxErr blk))
+           DeserialiseFailure m LB.ByteString
+localTxSubmissionCodec proxy =
+  codecLocalTxSubmission
+    nodeEncodeGenTx
+    nodeDecodeGenTx
+    (nodeEncodeApplyTxError proxy)
+    (nodeDecodeApplyTxError proxy)
+
+-- using ChainSync is a hack, and this should be rewritten with LocalStateQuery protocol
```

Comment:
Yeah, the feature is implemented, but we're waiting on network to enable it.

Reply:
Actually chain sync is the right one for getting the tip. To use the local state query, you already need to know a recent point on the chain.